app/uk/gov/hmrc/nisp/models/StatePensionExclusionFiltered.scala

```diff
@@ -17,19 +17,44 @@
 package uk.gov.hmrc.nisp.models
 
 import org.joda.time.LocalDate
-import play.api.libs.json.Json
-import uk.gov.hmrc.nisp.models.enums.Exclusion
+import play.api.libs.json.{Json, OFormat}
 import uk.gov.hmrc.time.TaxYear
 import play.api.libs.json.JodaReads._
 import play.api.libs.json.JodaWrites._
 
-case class StatePensionExclusionFiltered(exclusion: Exclusion.Exclusion,
-                                 pensionAge: Option[Int] = None,
-                                 pensionDate: Option[LocalDate] = None,
-                                 statePensionAgeUnderConsideration: Option[Boolean] = None) {
+
+trait StatePensionExclusionFilter {
+  def exclusion: Exclusion
+  def pensionAge: Option[Int]
+  def pensionDate: Option[LocalDate]
+  def statePensionAgeUnderConsideration: Option[Boolean]
   val finalRelevantStartYear: Option[Int] = pensionDate.map(TaxYear.taxYearFor(_).back(1).startYear)
 }
 
+trait CopeData {
+  def copeAvailableDate: LocalDate
+  def previousAvailableDate: Option[LocalDate]
+}
+
+case class StatePensionExclusionFiltered(
+                                 exclusion: Exclusion,
+                                 pensionAge: Option[Int] = None,
+                                 pensionDate: Option[LocalDate] = None,
+                                 statePensionAgeUnderConsideration: Option[Boolean] = None
+                              ) extends StatePensionExclusionFilter
+
+case class StatePensionExclusionFilteredWithCopeDate(
+                                                      exclusion: Exclusion,
+                                                      pensionAge: Option[Int] = None,
```

Comment:
Is pensionAge, pensionDate and statePensionAgeUnderConsideration needed for this scenario?